packages/inter-protocol/src/auction/auctionBook.js

```diff
@@ -0,0 +1,340 @@
+import '@agoric/zoe/exported.js';
+import '@agoric/zoe/src/contracts/exported.js';
+import '@agoric/governance/exported.js';
+
+import { M, makeScalarBigMapStore, provide } from '@agoric/vat-data';
+import { AmountMath, AmountShape } from '@agoric/ertp';
+import { Far } from '@endo/marshal';
+import { mustMatch } from '@agoric/store';
+import { observeNotifier } from '@agoric/notifier';
+
+import {
+  atomicRearrange,
+  ceilMultiplyBy,
+  floorDivideBy,
+  makeRatioFromAmounts,
+  multiplyRatios,
+  ratioGTE,
+} from '@agoric/zoe/src/contractSupport/index.js';
+import { E } from '@endo/captp';
+import { makeTracer } from '@agoric/internal';
+
+import { makeDiscountBook, makePriceBook } from './discountBook.js';
+import {
+  AuctionState,
+  isDiscountedPriceHigher,
+  makeBrandedRatioPattern,
+  priceFrom,
+} from './util.js';
+
+const { Fail } = assert;
+
+/**
+ * @file The book represents the collateral-specific state of an ongoing
+ * auction. It holds the book, the lockedPrice, and the collateralSeat that has
+ * the allocation of assets for sale.
+ *
+ * The book contains orders for the collateral. It holds two kinds of
+ * orders:
+ *   - Prices express the bid in terms of a Currency amount
+ *   - Discount  express the bid in terms of a discount (or markup) from the
+ *     most recent oracle price.
+ *
+ * Offers can be added in three ways. When the auction is not active, prices are
+ * automatically added to the appropriate collection. If a new offer is at or
+ * above the current price of an active auction, it will be settled immediately.
+ * If the offer is below the current price, it will be added, and settled when
+ * the price reaches that level.
+ */
+
+const trace = makeTracer('AucBook', false);
+
+/** @typedef {import('@agoric/vat-data').Baggage} Baggage */
+
+export const makeAuctionBook = async (
+  baggage,
+  zcf,
+  currencyBrand,
+  collateralBrand,
+  priceAuthority,
+) => {
+  const makeZeroRatio = () =>
+    makeRatioFromAmounts(
+      AmountMath.makeEmpty(currencyBrand),
+      AmountMath.make(collateralBrand, 1n),
+    );
```

Comment:
```suggestion
  const zeroRatio =
    makeRatioFromAmounts(
      AmountMath.makeEmpty(currencyBrand),
      AmountMath.make(collateralBrand, 1n),
    );
```
and then replace every `makeZeroRatio()` with `zeroRatio`.

Reply:
I noticed that somewhere else and removed it, but I apparently missed it here. I don't know what evolution the code must have gone through for that to have happened, but it sure looks dumb now.